wiki/Client/File_formats/Osu_(file_format)/en.md

```diff
@@ -4,346 +4,623 @@
-- `x` determines the index of the column that the hold will be in. It is computed by `floor(x * columnCount / 512)` and clamped between `0` and `columnCount - 1`.
-- `y` does not affect holds. It defaults to the centre of the playfield, `192`.
+`slides`
 
-### Examples
+> The slider is returnable if the value is more than `1`.
+
+`length`
+
+> If the slider's `length` is longer than the defined curve, the slider will extend until it reaches the target length:
+>
+> - For bézier, catmull, and linear curves, it continues in a straight line from the end of the curve
+> - For perfect circle curves, it continues the circular arc
+>
+> The slider's `length` can be used to determine the time it takes to complete the slider. `length / (SliderMultiplier*100)*beatLength` tells how many milliseconds it takes to complete one slide of the slider (assuming `beatLength` has been adjusted for inherited timing points).
+
+`edge_sounds`
+
+> The first sound is the one that plays when the slider is first clicked, and the last sound is the one that plays when the slider's end is hit.
+>
+> Apart from `edge_sounds`, sliders also have an ongoing hitsound whenever the player is in range of the slider's follow circle. The sound file is looped for as long as it is active. This hitsound uses the hit object's `hitSound` and `hitSample` properties, but only the normal and whistle sounds are supported. Its filename is `<sampleSet>-slider<hitSound><index>.wav`, where `hitSound` is either `slide` for normal or `whistle` for whistle.
+>
+> Besides the sounds of the beginning and the end `edge_sounds`, sliders also have an ongoing hitsound whenever the player is in range of the slider's follow circle. The sound file is looped for as long as it is active. This hitsound uses the hit object's `hit_sound` and `hit_sample`, but only the normal and whistle sounds are supported. Its filename is `<sampleSet>-slider<hitSound><index>.wav`, where `hitSound` is either `slide` for normal or `whistle` for whistle.
+
+### Spinner
+
+#### Parameters
+
+`x,y,start_time,object_type,hit_sound,end_time,hit_sample`
+
+| Option | Value type | Description | Value |
+| :-- | :-- | :-- | :-- |
+| `x` | Integer | The location of the spinner on the X axis in pixels | `0` not used |
+| `y` | Integer | The location of the spinner on the Y axis in pixels | `0` not used |
+| `start_time` | Integer | Time in milliseconds when the spinner should start | `1279` |
+| `object_type` | Integer | Bit flags indicating the type of object | bit `3` = `1` constant |
+| `hit_sound` | Integer | Bit flags indicating the hit sound applied to the object | `0` |
+| `end_time` | Integer | Time in milliseconds when the spinner should end | `1279` |
+| `hit_sample` | Colon-separated list | Information about which samples are played when the object is hit | `0:0:0:0` by default |
+
+#### Examples
+
+`0,0,1250,8,0,3250,0:0:0:0:`
+
+`0,0,2750,12,2,5000,0:0:0:0:`
```

Comment:
examples without any sort of accompanying information is quite useless (you can already read the syntax specification). ideally there'd be an image